tutorials/plugins/gdnative/gdnative-cpp-scons.rst

```diff
@@ -0,0 +1,872 @@
+
+The C++ bindings library you created at the start of the previous tutorial was a static library (.lib for Windows, .a for Linux/Mac). We will now need to include it when building our dynamic library.
+
+The name of the file should be `godot-cpp.<platform>.<bits>`. That exact name is not strictly necessary, but that is the format of the library currently exported by the godot-cpp repository's ``SConstruct`` file, so we'll need to match its format if we intend to locate it properly. Because we know we will have to include the C++ bindings, we can create a dedicated parameter for the path to it.
+
+.. code-block:: python
+
+    cpp_bindings_library_path = ARGUMENTS.get("cpp_bindings_library", "godot-cpp/bin/godot-cpp")
+    
+    platform = ARGUMENTS.get("platform", "windows")
+    bits = ARGUMENTS.get("bits", 64)
+
+    cpp_bindings_library = cpp_bindings_library_path + "." + platform + "." + str(bits)
+
+    env.Append(LIBS=[cpp_bindings_library])
+
+This results in users passing the file path to their bindings library into the scons build file. There isn't much sense in creating a separate parameter for the name of the library ("godot-cpp" here), so we can just include the library name in the path.
+
+However, we will have to grab a library that uses the same platform and bits compilation as we intend to produce in our dynamic library. As such, we will manually inject that part ourselves. After all, there's no sense including a Windows 32-bit static library when trying to build a MacOS 64-bit dynamic library, etc. To prepare for the possibility that our users will provide the full filepath though, rather than just up to the name of the library (or by mistake, supply the wrong extensions in their filepath!), we will strip away the extensions they provide and use only the actual name of the library.
+
+.. code-block:: python
+    import os
+
+    cpp_bindings_library_path = ARGUMENTS.get("cpp_bindings_library", "godot-cpp/bin/godot-cpp")
+    
+    # repeatedly divide out extensions until the extension string is empty.
+    ext = "z"
+    while ext != "":
+        # "godot-cpp.lib" would get split here into "godot-cpp" in the first value and ".lib" in the second
+        cpp_bindings_library_path, ext = os.path.splitext(cpp_bindings_library_path)
+    
+    platform = ARGUMENTS.get("platform", "windows")
+    bits = ARGUMENTS.get("bits", 64)
+
+    # manually setup our own appropriate file extensions
+    final_cpp_bindings_library_path = cpp_bindings_library_path + "." + platform + "." + str(bits)
+
+Now, if our users were to use the parameter ``cpp_bindings_library=godot-cpp/bin/godot-cpp.64.windows``, we can properly convert it to the expected format of <name>.<platform>.<bits> instead where <name> is "godot-cpp" in this case.
+
+That will give us the full filepath to our library. If we want to include other libraries, we would need to define another parameter. Perhaps a comma-separated list of filepaths?
```

Comment:
I think the final sentence here needs a bit more confidence.

"A simple comma-separated list of filepaths should suffice:"